internal/context.go

```diff
@@ -284,3 +322,90 @@ func createContextFileIfNotExist() {
 		log.Verboseln("Created ~/.corectl/contexts.yml for storage of corectl contexts")
 	}
 }
+
+func getSessionCookie(tlsClientConfig *tls.Config, engineURL string, userName string, password string) string {
+	// Verify Qlik Sense URL
+	u, err := url.Parse(engineURL)
+
+	if err != nil {
+		log.Fatalln("The engineURL doesn't seem to be correct")
+	}
+
+	if u.Scheme != "https" {
+		log.Fatalln("We only support login through secure connections (HTTPS)")
+	}
+
+	// Get username
+	if len(userName) == 0 {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print("Enter Username (domain\\user): ")
+		userName, _ = reader.ReadString('\n')
+	}
+
+	if !strings.Contains(userName, "\\") {
+		log.Fatalln("username MUST be in format 'domain\\user'")
+	}
+
+	// Get password
+	if len(password) == 0 {
+		fmt.Print("Enter Password: ")
+		bytePassword, _ := terminal.ReadPassword(int(syscall.Stdin))
+		password = string(bytePassword)
+	}
+
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsClientConfig
+
+	// Get post URL via redirects
+	resp, err := http.Get(engineURL)
+
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Generate xrfkey
+	xrfkey := generateXrfkey()
+
+	//
+	loginURL := resp.Request.URL
+	q, _ := url.ParseQuery(loginURL.RawQuery)
+	q.Add("xrfkey", xrfkey)
+	loginURL.RawQuery = q.Encode()
+
+	urlData := url.Values{}
+	urlData.Set("username", userName)
+	urlData.Set("pwd", password)
+
+	hc := http.Client{}
+	req, err := http.NewRequest("POST", loginURL.String(), strings.NewReader(urlData.Encode()))
+
+	req.PostForm = urlData
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("x-qlik-xrfkey", xrfkey)
+
+	postResp, postErr := hc.Do(req)
+
+	if postErr != nil {
+		log.Fatalln(postErr)
+	}
+
+	setCookie := postResp.Header.Get("Set-Cookie")
+
+	if setCookie == "" {
+		log.Fatalln("We wasn't able to get the 'X-Qlik-Session' cookie, please check your password.")
+	}
+
+	return strings.TrimRight(strings.Fields(setCookie)[0], ";")
+}
+
+func generateXrfkey() (xrfkey string) {
```

Comment:
If we are going to use named returns it should be consistently used, which I don't think it is :)

Reply:
Changed